### PR TITLE
[Snapshot-1544] Update the Tag snapshot

### PR DESCRIPTION
**Pull request by robergro**
Here the new strategy for the Snapshot !

This work was done with Elisa.
The link of the associated Google sheet: [here](https://docs.google.com/spreadsheets/d/1IfF0v9AWvDdqcYxD6l5cWqqNiUsBM3HGlRNkpWgSfGA/edit#gid=0)

The new tests correspond to the list of tests define inside the Google sheet.

We have 5 scenarios for the Tag.

At the end, we have now 39 images for each framworks instead of 486...

The PR for the Snapshot testing is [here](https://github.com/adevinta/spark-ios-snapshots/pull/65).

### CODE REVIEW

File: core/Sources/Components/Tag/View/Common/TagSutSnapshotTests.swift [hunk 1]
```diff
@@ -6,40 +6,35 @@
 //  Copyright © 2023 Adevinta. All rights reserved.
 //
 
+import UIKit
 @testable import SparkCore
 
 struct TagSutSnapshotTests {
```
Comment (michael-zimmermann) [inline core/Sources/Components/Tag/View/Common/TagSutSnapshotTests.swift:12]: The name of this struct is strange.
SUT stands for Service Under Tests and is usually given to the variable name of the service being tested.

File: core/Sources/Components/Tag/View/SwiftUI/TagViewSnapshotTests.swift [hunk 2]
```diff
@@ -17,57 +17,30 @@ final class TagViewSnapshotTests: SwiftUIComponentSnapshotTestCase {
...
-                matching: view,
-                testName: sut.testName()
-            )
+    func test() {
+        let scenarios = TagScenarioSnapshotTests.allCases
+
+        for scenario in scenarios {
+            let suts = scenario.sut(isSwiftUIComponent: true)
+            for sut in suts {
+                let view = TagView(theme: self.theme)
```
Comment (michael-zimmermann) [inline core/Sources/Components/Tag/View/SwiftUI/TagViewSnapshotTests.swift:29]: In this case the view is the SUT and not the variations.

File: core/Sources/Components/Tag/View/Common/TagScenarioSnapshotTests.swift [hunk 3]
```diff
@@ -0,0 +1,196 @@
...
+    ///
+    /// Description: To test all intents
+    ///
+    /// Content:
+    ///  - intents: all
+    ///  - variant: tinted
+    ///  - content: icon + text
+    ///  - mode: all
+    ///  - size: default
+    private func test1(isSwiftUIComponent: Bool) -> [TagSutSnapshotTests] {
```
Comment (michael-zimmermann) [inline core/Sources/Components/Tag/View/Common/TagScenarioSnapshotTests.swift:53]: The name of the function could be more descriptive, something like is in the comment.
Maybe the could be renamed to something like "scenarioAllIntents" .. etc. When I see a function starting with "test", I immediately assume, that it is a test.

File: core/Sources/Components/Tag/View/Common/TagScenarioSnapshotTests.swift [hunk 4]
```diff
@@ -0,0 +1,196 @@
...
+//  Created by robin.lemaire on 10/10/2023.
+//  Copyright © 2023 Adevinta. All rights reserved.
+//
+
+@testable import SparkCore
+import UIKit
+import SwiftUI
+
+enum TagScenarioSnapshotTests: String, CaseIterable {
+    case test1
```
Comment (michael-zimmermann) [inline core/Sources/Components/Tag/View/Common/TagScenarioSnapshotTests.swift:14]: The name of the scenarios could be more descriptive, see below.

File: core/Sources/Components/Tag/View/Common/TagSutSnapshotTests.swift (hunk 1 above)
Comment (robergro) [reply core/Sources/Components/Tag/View/Common/TagSutSnapshotTests.swift]: I replace SUT to configuration

**Review by michael-zimmermann — APPROVED** [review]

**Pull request merged**